subsys/bluetooth/audio/ascs.c

```diff
@@ -1158,12 +1158,18 @@ static void ase_stream_add(struct bt_ascs *ascs, struct bt_ascs_ase *ase,
 	stream->ep = &ase->ep;
 }
 
+static void ascs_init(struct bt_ascs *ascs, struct bt_conn *conn)
+{
+	ascs->conn = bt_conn_ref(conn);
+	sys_slist_init(&ascs->ases);
+}
+
```

Comment:
> The fact that ascs is taken from global variable that is guaranteed to be initialized to zero is implementation specific.

How do you mean? 
The C standard requires all global variables to be zero initialized (and has been that way since C89/ANSI C):

> If an object that has static storage duration is not initialized explicitly, it is initialized implicitly as if every member that has arithmetic type were assigned 0 and every member that has pointer type were assigned a null pointer constant.

> If future decision would be made to take it from other memory, eg. slab, the uninitialized values would lead to unexpected behaviors and hard to trace bugs.

Correct, if we get the memory from somewhere else, then the above may no longer be true. However, that can be said about and and all the global values we have (and not just in ASCS), and a lot of the code base builds on that fact. 

I'm not against this change at all, but just saying that currently this doesn't change any behavior

Reply:
> > The fact that ascs is taken from global variable that is guaranteed to be initialized to zero is implementation specific.
> 
> How do you mean? The C standard requires all global variables to be zero initialized (and has been that way since C89/ANSI C):

Implementation specific is that it takes it from global variable, that's what I meant